Index: Texinfo/ModulePath.pm.in
===================================================================
--- Texinfo/ModulePath.pm.in	(revision 8063)
+++ Texinfo/ModulePath.pm.in	(working copy)
@@ -73,6 +73,7 @@
     if ('@USE_EXTERNAL_UNIDECODE@' ne 'yes') {
       unshift @INC, (File::Spec->catdir($lib_dir, 'lib', 'Text-Unidecode', 'lib'));
     }
+    unshift @INC, (File::Spec->catdir($lib_dir, '..', 'parsetexi'));
   }
 
   if (!$libexec_dir) {
Index: Texinfo/Parser.pm
===================================================================
--- Texinfo/Parser.pm	(revision 8058)
+++ Texinfo/Parser.pm	(working copy)
@@ -27,6 +27,11 @@
 #  user visible subroutines.
 #  internal subroutines, doing the parsing.
 
+
+use Parsetexi;
+1;
+__END__
+
 package Texinfo::Parser;
 
 # We need the unicode stuff.
Index: texi2any.pl
===================================================================
--- texi2any.pl	(revision 8058)
+++ texi2any.pl	(working copy)
@@ -262,8 +262,8 @@
   } else {
     # used in the standalone perl module, as $hardcoded_version is undef
     # and it should never be configured in that setup
-    require Texinfo::Parser;
-    $configured_version = $Texinfo::Parser::VERSION;
+    ## require Texinfo::Parser;
+    ## $configured_version = $Texinfo::Parser::VERSION;
   }
 }
 
@@ -1065,7 +1065,12 @@
   document_warn(__('--Xopt option without printed output')); 
 }
 
-require Texinfo::Parser;
+#require Texinfo::Parser;
+BEGIN {
+require Parsetexi;
+*Texinfo::Parser:: = \%Parsetexi::;
+}
+
 require Texinfo::Structuring;
 require Texinfo::Transformations;
 # Avoid loading these modules until down here to speed up the case
